### PR TITLE
Signing transactions through wallet instance

### LINKED ISSUE #35: Implement Wallet.signTransaction
For the successful completion of this task, a wallet instance must be able to do `wallet.signTransaction(tx)` and produce a legit transaction signature (string) that can be sent to the provider (consensus nodes) for submission.

1. The `TransactionRequest` type must be changed. `nonce`, `gasPrice`, `type`, `accessList`, `maxPriorityFeePerGas`, and `maxFeePerGas` properties must be removed. `from` field should contain either `evmAddress` or hedera `account` that will be populated in `TransactionID.accountID` field.

### PR DESCRIPTION
**Pull request by joan41868**
**Description**:

This **PR** provides support for signing Hedera transactions with the `Wallet.signTransaction(txRequest)` method.
The current implementation will sign only one of 4 types of transactions:
- FileCreate
- FileAppend
- ContractCreate
- ContractCall

**Related issue(s)**:

Fixes #35 

**Notes for reviewer**:
There is a JSDoc which explains the conditions required in order to understand the function output.
Tests are provided, and those tests include typecasting in order to faciliate the operation comprehension.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: examples/wallet.example.ts [hunk 1]
```diff
@@ -86,4 +90,29 @@ const hethers = require("ethers");
...
+	const txData = await new TransferTransaction()
+		.setTransactionId(TransactionId.generate("0.0.98"))
+		.setNodeAccountIds([new AccountId(0, 0, 3)])
+		.freeze()
+		.toBytesAsync();
+	const tx = {
+		to : getAddressFromAccount("0.0.98"),
+		from: "",
+		gasLimit: 0,
+		data: hexlify(txData),
```
Comment (Daniel-K-Ivanov) [inline examples/wallet.example.ts:103]: this should not be the `data` property. Data property is Ethereum is `{functionSignature+arguments}`

File: examples/wallet.example.ts [hunk 2]
```diff
@@ -86,4 +90,29 @@ const hethers = require("ethers");
...
+		value : 0,
+		chainId: 292,
+		type : 1,
+		customData: {
+			abi:{
+
+			},
+			values:{
+
+			}
```
Comment (Daniel-K-Ivanov) [inline examples/wallet.example.ts:113]: This is actually the data encoded in `data` property

**Review by Daniel-K-Ivanov — COMMENTED** [review]
I think that we are in the right direction 🙌  Added a couple of comments

File: packages/wallet/src.ts/index.ts [hunk 3]
```diff
@@ -1,266 +1,338 @@
...
+     * - ContractCall - when there is a `to` field present. Ignores the other fields
+     *
+     * @param transaction - the transaction to be signed.
+     */
+    signTransaction(transaction: TransactionRequest): Promise<string> {
+        let tx: Transaction;
+        let arrayifiedData :Uint8Array;
+        if (transaction.data) {
+            arrayifiedData = arrayify(transaction.data);
+        }
```
Comment (Daniel-K-Ivanov) [inline packages/wallet/src.ts/index.ts:214]: nit: Can we use always `arrayify` even if `.data` is empty in order to skip the `if`

File: packages/wallet/src.ts/index.ts [hunk 4]
```diff
@@ -1,266 +1,338 @@
...
+        const gas = BigNumber.from(transaction.gasLimit).toNumber();
+        if (transaction.to) {
+            tx = new ContractExecuteTransaction()
+                .setContractId(ContractId.fromSolidityAddress((transaction.to.toString())))
```
Comment (Daniel-K-Ivanov) [inline packages/wallet/src.ts/index.ts:218]: nit: remove unnecessary `( )`

File: packages/wallet/src.ts/index.ts [hunk 5]
```diff
@@ -1,266 +1,338 @@
...
+                .setFunctionParameters(arrayifiedData)
```
Comment (Daniel-K-Ivanov) [inline packages/wallet/src.ts/index.ts:219]: nit: we can do `transaction.data ? arrayify(transaction.data) : []` and skip the if above

File: packages/wallet/src.ts/index.ts (hunk 3 above)
Comment (joan41868) [reply packages/wallet/src.ts/index.ts:214]: Agree, but it must be present as a property ( `!undefined` ) in order for `arrayify` to not throw argument error. This is what the if stands for

Comment (joan41868) [reply packages/wallet/src.ts/index.ts:214]: Skipped, nevermind the previous comment.

**Pull request closed without merging**